packages/extension-base/src/background/handlers/Extension.ts

```diff
@@ -334,14 +336,18 @@ export default class Extension {
     };
   }
 
-  private async signingApprovePassword ({ id, password, savePass }: RequestSigningApprovePassword, getContentPort: GetContentPort): Promise<void> {
+  private async signingApprovePassword ({ id, password: inputPassword, savePass }: RequestSigningApprovePassword, getContentPort: GetContentPort): Promise<void> {
     const queued = await this.#state.getSignRequest(id);
 
     assert(queued, 'Unable to find request');
 
     const { payload } = queued;
     const pair = keyring.getPair(queued.account.address);
 
+    const { cachedPassword } = await this.refreshAccountPasswordCache(pair);
```

Comment:
my idea was to call `refreshAccountPasswordCache` once. I think this should work?

Reply:
I'm not sure, I think we may need to do it in multiple places as it is

Reply:
why?

Reply:
My bad, I just realised what  you were talking about